### PR TITLE
Add support for Python 3.6

### PR DESCRIPTION
**Pull request by jamadden**
Also add support for pip caching on Travis, and DRY with dependencies
in tox.ini.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,11 +5,17 @@ python:
   - 3.3
   - 3.4
   - 3.5
-  - pypy
-  - pypy3
+  - 3.6
+  - pypy-5.4.1
```
Comment (mgedmin) [inline .travis.yml]: No `pypy3.3-5.2-alpha1`?

File: .travis.yml [hunk 2]
```diff
@@ -5,11 +5,17 @@ python:
...
 script:
-  - tox
+  - zope-testrunner --test-path=src  --auto-color --auto-progress
 notifications:
     email: false
+after_success:
+    - coveralls
+cache: pip
+before_cache:
+    - rm -f $HOME/.cache/pip/log/debug.log
```
Comment (mgedmin) [inline .travis.yml]: The indentation seems rather inconsistent here.

Before I think we had a strange but still somewhat coherent style of

```
item:
  - list
  - item
item:
    dict: item
```

but now it's all over the place :/

File: .travis.yml (hunk 1 above)
Comment (jamadden) [inline .travis.yml]: pypy3.3 won't ever get beyond alpha status AFAIU (plus general python 3.3 support ends soon too). The pypy3 devs are focusing all of their attention on pypy3.5, which is currently in beta. It's possible to test that on travis, but it takes some work.

Given the presence of a pypy2 and a python 3.3, and a python 3.5, I think we have the possible combos covered pretty well, with the exception of alpha- or beta-specific bugs, and given how fast pypy dev is moving now, it's probably not really worth adding workarounds for those, at least in my opinion.

File: .travis.yml (hunk 2 above)
Comment (jamadden) [inline .travis.yml]: Outside of existing lists, where I try to preserve the existing indentation, I just hit enter or tab and do what Emacs's YAML mode says by default. I'm *sure* you can agree that that's the authoritative source 😉

**Comment by jamadden** [comment]
Thanks! A PyPI release will be helpful to move zopefoundation/zope.file#1 along.

**Comment by mgedmin** [comment]
You now have PyPI rights for zope.publisher.

**Pull request merged**